src/monattack.cpp

```diff
@@ -1650,6 +1650,64 @@ void mattack::leap(monster *z, int index)
     }
 }
 
+void mattack::impale(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 1 ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+
+    z->reset_special(index); // Reset timer
+    z->moves -= 80;
+    bool uncanny = foe != nullptr && foe->uncanny_dodge();
+    if( uncanny || dodge_check(z, target) ){
+    if( foe != nullptr ) {
+        if( seen ) {
+            auto msg_type = foe == &g->u ? m_warning : m_info;
+            foe->add_msg_player_or_npc( msg_type, _("The %s lunges at you, but you dodge!"),
+                                                _("The %s lunges at <npcname>, but they dodge!"),
+                                            z->name().c_str() );
+        }
+        if( !uncanny ) {
+            foe->practice( "dodge", z->type->melee_skill * 2 );
+            foe->ma_ondodge_effects();
+        }
+    } else if( seen ) {
+        add_msg( _("The %s lunges at %s, but misses!"), z->name().c_str(), target->disp_name().c_str() );
+    }
+    return;
+    }
+    foe->apply_damage( z, bp_torso, rng(15, 20) );
```

Comment:
Not sure if direct damage is proper here. This bypasses power armors, not just survivor gear.
Look up damage.h - it has the ability to add flat armor penetration and armor multiplier (flat is applied first). That way you could have heavy armor give a small bonus, just not worth the dodge penalties. Heavy power armor should still stop the impale, because it is pretty much like wearing a tank.